src/control/server/mgmt_pool.go

```diff
@@ -506,6 +506,53 @@ func (svc *mgmtSvc) PoolDestroy(ctx context.Context, req *mgmtpb.PoolDestroyReq)
 		req.SvcRanks = system.RanksToUint32(ps.Storage.CreationRanks())
 		inCleanupMode = true
 	} else {
+		req.SvcRanks = system.RanksToUint32(ps.Replicas)
+	}
+
+	// Pool handle eviction step: perform separate drpc.MethodPoolEvict.
+	// Do this _before_ transitioning the pool to PoolServiceStateDestroying.
+	// Upon -DER_BUSY, keep pool in its current state and reply with the error immediately.
+	evreq := &mgmtpb.PoolEvictReq{}
+	evreq.Sys = req.Sys
+	evreq.Id = req.Id
+	evreq.SvcRanks = req.SvcRanks
+	evreq.Destroy = true
+	evreq.Force = req.Force
+	svc.log.Debugf("MgmtSvc.PoolDestroy issuing drpc.MethodPoolEvict, evreq:%+v\n", evreq)
+	dresp, err := svc.makePoolServiceCall(ctx, drpc.MethodPoolEvict, evreq)
+	if err != nil {
+		return nil, err
+	}
+
+	evresp := &mgmtpb.PoolEvictResp{}
+	if err = proto.Unmarshal(dresp.Body, evresp); err != nil {
+		return nil, errors.Wrap(err, "unmarshal PoolEvict response")
+	}
+
+	svc.log.Debugf("MgmtSvc.PoolDestroy drpc.MethodPoolEvict, evresp:%+v\n", evresp)
+
+	ds := drpc.DaosStatus(evresp.Status)
+	switch ds {
+	case drpc.DaosSuccess:
+		break
+	case drpc.DaosNotLeader, drpc.DaosNotReplica:
+		if !inCleanupMode {
+			// If we're not cleaning up, then this is an error.
+			svc.log.Errorf("PoolDestroy dRPC call failed due to %s in non-cleanup path", ds)
+		}
+		break
+	default:
+		svc.log.Errorf("PoolEvict (first step of destroy) dRPC call failed: %s", ds)
+	}
```

Comment:
Would probably be better to refactor this and the other switch below into a common helper, e.g. `func checkStatus(cleanup bool, status drpc.DaosStatus) error` ...

You can make the helper private to this method by declaring it as a variable, e.g. `checkStatus := func(...)`

Reply:
reworked the logic in the latest version of the patch so that the code does not have two switch statements.